portal_user_account_deactivate/controllers/portal.py

```diff
@@ -13,5 +13,12 @@ def deactivate_account(self, **post):
         user = request.env.user
         if user:
             user.with_user(SUPERUSER_ID).write({"active": False})
```

Comment:
Yes. We need it because we can't archive partner_id before we archive user. We can't also delete user instead of archive in this step before updating partner record because there is some validation in `phone_validation` using current user.